Drv/LinuxGpioDriver/LinuxGpioDriver.cpp

```diff
@@ -0,0 +1,314 @@
+    Os::File::Status status = Os::File::Status::OTHER_ERROR;
+    switch (errno_input) {
+        case 0:
+            status = Os::File::Status::OP_OK;
+            break;
+        case EBADF:
+            status = Os::File::Status::NOT_OPENED;
+            break;
+        case EINVAL:
+            status = Os::File::Status::INVALID_ARGUMENT;
+            break;
+        case ENODEV:
+            status = Os::File::Status::DOESNT_EXIST;
+            break;
+        case ENOMEM:
+            status = Os::File::Status::NO_SPACE;
+            break;
+        case EPERM:
+            status = Os::File::Status::NO_PERMISSION;
+            break;
+        case ENXIO:
+            status = Os::File::Status::INVALID_MODE;
+            break;
+        // Cascades intended
+        case EFAULT:
+        case EWOULDBLOCK:
+        case EBUSY:
+        case EIO:
+        default:
+            status = Os::File::Status::OTHER_ERROR;
+            break;
+    }
+    return status;
+}
+
+Drv::GpioStatus errno_to_gpio_status(PlatformIntType errno_input) {
+    Drv::GpioStatus status = Drv::GpioStatus::T::UNKNOWN_ERROR;
+    switch (errno_input) {
+        case EBADF:
+            status = Drv::GpioStatus::T::NOT_OPENED;
+            break;
+        case ENXIO:
+            status = Drv::GpioStatus::INVALID_MODE;
+            break;
+        // Cascades intended
+        case EFAULT:
+        case EINVAL:
+        case EWOULDBLOCK:
+        case EBUSY:
+        case EIO:
+        default:
+            status = Drv::GpioStatus::T::UNKNOWN_ERROR;
+            break;
+    }
+    return status;
+}
+
+U32 configuration_to_handler_flags(Drv::LinuxGpioDriver::GpioConfiguration configuration) {
+    U32 flags = 0;
+    switch (configuration) {
+        case LinuxGpioDriver::GPIO_OUTPUT:
+            flags = GPIOHANDLE_REQUEST_OUTPUT;
+            break;
+        // Cascade intended
+        case LinuxGpioDriver::GPIO_INPUT:
+        case LinuxGpioDriver::GPIO_INTERRUPT_RISING_EDGE:
+        case LinuxGpioDriver::GPIO_INTERRUPT_FALLING_EDGE:
+            flags = GPIOHANDLE_REQUEST_INPUT;
+            break;
+        default:
+            FW_ASSERT(0);
+            break;
+    }
+    return flags;
+}
+
+U32 configuration_to_event_flags(Drv::LinuxGpioDriver::GpioConfiguration configuration) {
+    U32 flags = 0;
+    switch (configuration) {
+        case LinuxGpioDriver::GPIO_INTERRUPT_RISING_EDGE:
+            flags = GPIOEVENT_REQUEST_RISING_EDGE;
+            break;
+        case LinuxGpioDriver::GPIO_INTERRUPT_FALLING_EDGE:
+            flags = GPIOEVENT_REQUEST_FALLING_EDGE;
+            break;
+        case LinuxGpioDriver::GPIO_INTERRUPT_BOTH_RISING_AND_FALLING_EDGES:
+            flags = GPIOEVENT_REQUEST_RISING_EDGE | GPIOEVENT_REQUEST_FALLING_EDGE;
+            break;
+        default:
+            FW_ASSERT(0);
+            break;
+    }
+    return flags;
+}
+
+LinuxGpioDriver ::~LinuxGpioDriver() {
+    ::close(this->m_fd);
```

Comment:
Do you want to check if `this->m_fd` is valid? Otherwise closing an invalid file descriptor will result in an `EBADF` errno. Perhaps we expect the destructor to be called when FSW is rebooting; so, we don't really care.

Should `(void)` out the return of this call.